service/history/queue/transfer_queue_processor_base.go

```diff
@@ -66,12 +68,14 @@ type (
 		metricsClient metrics.Client
 		metricsScope  metrics.Scope
 
-		rateLimiter  quotas.Limiter
-		lastPollTime time.Time
-		notifyCh     chan struct{}
-		status       int32
-		shutdownWG   sync.WaitGroup
-		shutdownCh   chan struct{}
+		rateLimiter   quotas.Limiter
+		notifyCh      chan struct{}
```

Comment:
With the poll time mechanism, do we still need this notifyCh?

Reply:
I think we still need this notifyCh for two reasons:
- Without it, each notifyNewTask call need to grab a lock on `nextPollTime` which may be blocking
- Without the notifyCh, we will need to update the poll timer for every new task, which involves starting/stopping/resetting timer and can be expensive.